Microsoft.HealthVault/ItemTypes/ApproximateDateTime.cs

```diff
@@ -542,6 +542,11 @@ public int CompareTo(DateTime other)
         ///
         public override bool Equals(object obj)
         {
+            if (obj?.GetType() != this.GetType())
```

Comment:
Good point. I was just transplanting a fix but this is opening a new can of worms. Made it accept DateTime and fixed a bug where CompareTo was failing due to a missing method on ApproximateTime. Also added more tests and fixed another bug in HealthServiceDateTime.

---
In reply to: [117345820](https://github.com/Microsoft/healthvault-dotnetstandard-sdk/pull/36#discussion_r117345820) [](ancestors = 117345820)